crates/snapbox/src/assert.rs

```diff
@@ -197,22 +197,15 @@ impl Assert {
     ) -> (crate::Result<crate::Data>, crate::Data) {
         let expected = expected.map(|d| d.normalize(NormalizeNewlines));
         // On `expected` being an error, make a best guess
-        if let (Some(expected), format) = expected
-            .as_ref()
-            .map(|d| (d.render(), d.format()))
-            .unwrap_or((Some("".to_string()), DataFormat::Text))
-        {
-            actual = actual.try_coerce(format);
-            if self.normalize_paths {
-                actual = actual.normalize(NormalizePaths);
-            }
-            actual = actual
-                .normalize(NormalizeNewlines)
-                .normalize(NormalizeMatches::new(
-                    &self.substitutions,
-                    &crate::Data::text(expected),
-                ));
+
+        let ok = expected.clone().unwrap_or_default();
```

Comment:
Why is it cloned?

Why are we defaulting the whole thing?

Reply:
`clone` is because we can't use `as_ref` since it would make the `unwarp` portion need to be ref as well. As for `unwrap_or_default` I believe it would make it `Data::new()` which is `Text` but I can see now that an explicit `unwrap_or(Data::new())` might be the better idea.

Reply:
> clone is because we can't use as_ref since it would make the unwarp portion need to be ref as well.

You could create a dummy variable and unwrap to that.

but really, this is contingent on the unwrap

> As for unwrap_or_default I believe it would make it Data::new() which is Text but I can see now that an explicit unwrap_or(Data::new()) might be the better idea.

If instead we do `expected.as_ref().map(|e| e.format()).unwrap_or_default())` and conditionalize the `NormalizeMatches` on there being an `expected`, then we don't need the clone.  The performance of it probably doesn't matter but this would help clarify intent.